### DISCUSSION TITLE
windowrulev2 does not work consistently

### DISCUSSION THREAD

**Discussion by gmr458** [hyprwm/Hyprland#3903] [Q&A]
I Create some window rules for the file save or file upload windows, these windows always open too large, the rules are to set the size to 700x500 and center the window, it works, but sometimes it doesn't work, here is a screen recording.

https://github.com/hyprwm/Hyprland/assets/53343401/be6cdd9d-55a8-4a17-920a-c91d172bfbb1

```
windowrulev2 = size 700 500,class:(Xdg-desktop-portal-gtk),title:(Open File)
windowrulev2 = center,class:(Xdg-desktop-portal-gtk),title:(Open File)

windowrulev2 = size 700 500,class:(Xdg-desktop-portal-gtk),title:(All File)
windowrulev2 = center,class:(Xdg-desktop-portal-gtk),title:(All File)
```

**Comment by imnotpoz** [comment]
on my system, the gtk portal's class is `xdg-desktop-portal-gtk` and window rules (both v1 and v2) are case sensitive — try using that instead
